spec/formatting.md

```diff
@@ -0,0 +1,20 @@
+# WIP DRAFT MessageFormat 2.0 Formatting Behaviour
+
+## Introduction
+
+This document defines the behaviour of a MessageFormat 2.0 implementation
+when formatting a message for display in a user interface, or for some later processing.
+
+The document is part of the MessageFormat 2.0 specification,
+the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
+
+## Variable Resolution
+
+To resolve the value of a Variable,
+its Name is used to identify either a local variable,
+or a variable defined elsewhere.
+If a local variable and an externally defined one use the same name,
+the local variable takes precedence.
+
+It is an error for a local variable definition to
```

Comment:
The language is a bit awkward; I suggest:

Any reference to a local variable before it is defined is an error.

Reply:
@macchiati Good point; replaced with your suggestion.